extensions/amp-ventuno-player/0.1/amp-ventuno-player.js

```diff
@@ -0,0 +1,339 @@
+
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    const src = this.getVideoIframeSrc_();
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+
+    this.iframe_ = iframe;
+    this.element.appendChild(this.iframe_);
+
+    const loaded = this.loadPromise(this.iframe_).then(() => {
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+    });
+    this.playerReadyResolver_(loaded);
+    return loaded;
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+    return true; // Call layoutCallback again.
+  }
+
+  /** @override */
+  pauseCallback() {
+    this.pause();
+  }
+
+  getPlayerType_() {
+    return user().assert(
+        this.element.getAttribute('data-player'),
+        'The data-player attribute is required for <amp-ventuno-player> %s',
+        this.element);
+  }
+
+  getPubid_() {
+    return user().assert(
+        this.element.getAttribute('data-pubid'),
+        'The data-pubid attribute is required for <amp-ventuno-player> %s',
+        this.element);
+  }
+
+  getSlotid_() {
+    return user().assert(
+        this.element.getAttribute('data-slotid'),
+        'The data-slotid attribute is required for <amp-ventuno-player> %s',
+        this.element);
+  }
+
+  /** @return {string} */
+  getVideoIframeSrc_() {
+    const pubid = encodeURIComponent(this.pubid_ || ''),
+        slotid = encodeURIComponent(this.slotid_ || ''),
+        pType = this.playerType_ || '',
+        optParams = {};
+
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+    // As of now, only the 'ep' player type will be supported
+    user().assert(this.playerType_ === 'ep',
+        'Only Editorial Player is supported');
+
+    dev().assert(this.pubid_);
+    dev().assert(this.slotid_);
+
+    let src = `https://venwebsecure.ventunotech.com/embed/embedPlayer.html?pFrom=amp&pType=${pType}&pubKey=${pubid}&slot=${slotid}`;
+
+    if (this.title_) {
+      optParams['pTitle'] = this.title_;
+    }
+
+    if (this.url_) {
+      optParams['pUrl'] = this.url_;
+    }
+
+    if (this.meta_) {
+      optParams['pMeta'] = this.meta_;
+    }
+
+    src = addParamsToUrl(src, optParams);
+
+    return this.videoIframeSrc_ = src;
+  }
+
+  /**
+   * Sends a command to the player through postMessage.
+   * @param {string} command
+   * @param {Array=} opt_args
+   * @private
+   */
+  sendCommand_(command) {
```

Comment:
this component needs to also receieve events from the iframe and dispatch `VideEvents` for pause, ended, etc.. See how amp-youtube does it here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-youtube/0.1/amp-youtube.js#L339